### DISCUSSION TITLE
OAuth2PasswordBearer swagger UI form not working properly

### DISCUSSION THREAD

**Discussion by ThinkalVB** [fastapi/fastapi#11416] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends
from fastapi.security import OAuth2PasswordBearer
from fastapi.security import OAuth2PasswordRequestForm

route_executive  = FastAPI()
oauth2_executive = OAuth2PasswordBearer(tokenUrl="token",
                                        scheme_name="Executive OAuth2")

#----------------------------------- Token Management ------------------------#
@route_executive.post("/token", status_code=201)
def token_creation(credentials: OAuth2PasswordRequestForm = Depends()):
    print(str(credentials.__dict__))
    print(credentials.client_id)        # Gives None when authenticating using OAuth2PasswordBearer

@route_executive.get("/token")
def token_details(access_token=Depends(oauth2_executive)):
    pass
```

### Description

Authentication form generated by OAuth2PasswordBearer have client_id and client_secret
However, during authentication, the client_id and client_secret passed to the form are not received at the server side.

### Operating System

Linux

### Operating System Details

Ubuntu 22.04.3 LTS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

Python 3.10.12

**Comment by YuriiMotov** [comment, marked as answer]
To receive this data as a Form data, you should select `Client credentials location` = `Request body` in the Swagger authorization form.
By default it's set to `Authorization header`.
![sw](https://github.com/tiangolo/fastapi/assets/109919500/e38c547d-0cb2-4698-9f28-22adb1b3126e)

**Reply by ThinkalVB** [reply]
Yes, that fixes the problem. Thank you.
